rig/machine_control/machine_controller.py

```diff
@@ -2433,6 +2446,16 @@ def close(self):
             self.flush()
             self.closed = True
 
+    def free(self):
+        """Free the memory represented by the file-like."""
```

Comment:
Would it be worth crashing out if you try and free a subregion? I can only imagine that resulting in bugs(?).

The alternative might be freeing once all subregions are freed but that seems like an invitation for complexity and memory leaks...